internal/tools/monitor/core/settings/retention-strategy/provider.go

```diff
@@ -29,10 +30,29 @@ import (
 	"github.com/erda-project/erda-infra/providers/httpserver/interceptors"
 )
 
+type TTL struct {
+	// hot data ttl
+	HotData time.Duration
+
+	// entire data ttl, cold data = ttl - hot ttl
+	All time.Duration
+}
+
+func (t TTL) GetHotTTLByDays() int64 {
+	return int64(math.Ceil(math.Max(t.HotData.Hours()/24, 1)))
+}
+
+func (t TTL) GetTTLByDays() int64 {
+	return int64(math.Ceil(math.Max(t.All.Hours()/24, 1)))
+}
+
 // Interface .
 type Interface interface {
 	GetTTL(key string) time.Duration
+	GetTTLv2(key string) *TTL
```

Comment:
you'd better to remove `GetTTL` method entirely, otherwise it will be more and more confused.